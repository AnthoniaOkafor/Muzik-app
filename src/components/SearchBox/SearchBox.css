@import "//netdna.bootstrapcdn.com/font-awesome/3.0/css/font-awesome.css";

input, input:focus, .searchBox::-webkit-input-placeholder {
    font-family: FontAwesome;
    background-color: #ECECEC;
    word-spacing: 0.2em ;
    padding: 20px 30px;
    color: white;
    opacity: 1;
    font-size: 17.4px;
    width: 86%;
    /* width: 350px; */
}


.search{
    display: flex;
}
.search > button {
    background-color: #1abc9c;
    color: white;
    border-radius: 0px 12px 12px 0px;
    border: 0px solid #1ABC9C;
    padding: 0px 31px;
    font-size: 15px;
    transition: all 0.3s ease;
}
.search > button:hover{
  border-color: #18b495;
  background-color: #18b495;
  cursor: pointer;
}

input:focus, .search > button:focus{
    outline: none;
}

.search > input {
    border: 0px solid #ECECEC;
    border-radius: 12px 0px 0px 12px;
}
