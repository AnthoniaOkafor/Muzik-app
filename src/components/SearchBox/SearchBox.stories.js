import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withKnobs, text, number } from "@storybook/addon-knobs";
import SearchBox from "./index";

storiesOf("SearchBox", module)
  .addDecorator(
    withInfo({
      text: "This is a SearchBox component that renders an input field and a button"
    })
  )
  .addDecorator(withKnobs())
  .add("SearchBox", () => (
    // write stories for functions
    <SearchBox onSearch={work} buttonTitle={text("buttonTitle", "button title")}/>

  ));
