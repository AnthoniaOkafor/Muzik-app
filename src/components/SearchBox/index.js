import React from 'react'
import PropTypes from "prop-types";
import "./SearchBox.css"

import Button from "../Button"

const SearchBox = ({onSearch}) =>{

	const handleSubmit = (e) =>{
		e.preventDefault()
		const searchInput = document.getElementById("box");
		onSearch(searchInput.value);
		searchInput.value = "";
}
	return(
		<form className="search" onSubmit={handleSubmit}>
			<input className="searchBox" id="box" type='text' placeholder='&#xf002; what event are you searching for?'/>
			<Button title={"search"} onClick={handleSubmit}/>
		</form>
	)
}

SearchBox.propTypes = {
  onSearch: PropTypes.func,
};	
export default SearchBox