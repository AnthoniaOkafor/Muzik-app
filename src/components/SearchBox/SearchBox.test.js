import React from "react";
import { mount } from "enzyme";

import SearchBox from "./index";
// check the button is rendered with the right title if title is passed 
// input renderes
const search = (value) => {
    console.log(value)
}

const setUp = (search, title) => {
    return mount(<SearchBox onSearch={search} buttonTitle={title}/>)
}
describe("SearchBox Component", () => {
  let searchBox;
  beforeEach(() => {
    searchBox = setUp(search, "button title");
  });

  it("should match snap shot", () => {

    expect(searchBox).toMatchSnapshot();
  });

  it("renders button with the correct title", () => {
    const button = searchBox.find("button");
    button.simulate("click", search("clicked"))

    expect(button.length).toBe(1);
    expect(button.text()).toEqual("button title");
  });

  it("renders input field", () => {
    const input = searchBox.find("input");

    expect(input.length).toBe(1);
  });
});