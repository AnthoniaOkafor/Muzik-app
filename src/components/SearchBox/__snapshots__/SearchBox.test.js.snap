// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component should match snap shot 1`] = `ReactWrapper {}`;
